app/models/katello/concerns/http_proxy_extensions.rb

```diff
@@ -17,16 +20,29 @@ def update_default_proxy_setting
         changes = self.previous_changes
         if changes.key?(:name)
           previous_name = changes[:name].first
-          setting = Setting::Content.unscoped.
-            where(name: "content_default_http_proxy").
-            first
+          setting = Setting.find_by(name: 'content_default_http_proxy')
 
           if setting && setting.value == previous_name && !previous_name.blank?
             setting.update_attribute(:value, self.name)
           end
         end
       end
 
+      def update_repository_proxy_details
+        changes = self.previous_changes
+        if changes.key?(:url) || changes.key?(:username) || changes.key?(:password)
+
+          repos =
+            Repository.joins(:root).where("#{RootRepository.table_name}.http_proxy_id = ?", id) +
+            Repository.joins(:root).where("#{RootRepository.table_name}.http_proxy_policy = ?", RootRepository::GLOBAL_DEFAULT_HTTP_PROXY)
+          repos.uniq.each do |repo|
+            ForemanTasks.async_task(
+              ::Actions::Katello::Repository::UpdateHttpProxyDetails,
+              repo)
+          end
+        end
```

Comment:
this isn't taking into account that the http proxy that was updated might not be the global proxy, we'd only want to update repos set to 'GLOBAL_DEFAULT_HTTP_PROXY' if the one being updated is the global proxy.